go-controller/pkg/ovn/ovn.go

```diff
@@ -225,84 +258,187 @@ func getPodNamespacedName(pod *kapi.Pod) string {
+		GenericController: GenericController{
+			client: ovnClient.KubeClient,
+			kube: &kube.Kube{
+				KClient:              ovnClient.KubeClient,
+				EIPClient:            ovnClient.EgressIPClient,
+				EgressFirewallClient: ovnClient.EgressFirewallClient,
+				CloudNetworkClient:   ovnClient.CloudNetworkClient,
+			},
+			watchFactory: wf,
+			recorder:     recorder,
+			nbClient:     libovsdbOvnNBClient,
+			sbClient:     libovsdbOvnSBClient,
+			podRecorder:  metrics.NewPodRecorder(),
+		},
+		defaultWg:       wg,
+		defaultStopChan: stopChan,
+		identity:        identity,
+	}
+}
+
+func (cm *ControllerManager) InitDefaultController(addressSetFactory addressset.AddressSetFactory) (*Controller, error) {
+	defaultNetConf := &ovncnitypes.NetConf{
+		NetConf: cnitypes.NetConf{
+			Name: ovntypes.DefaultNetworkName,
+		},
+		NetCidr:     config.Default.RawClusterSubnets,
+		MTU:         config.Default.MTU,
+		IsSecondary: false,
+	}
+	nadInfo, _ := util.NewNetAttachDefInfo(defaultNetConf)
+	return cm.NewOvnController(nadInfo, addressSetFactory)
+}
+
+func (cm *ControllerManager) Init() error {
+	if cm.defaultController != nil {
+		return nil
+	}
+
+	// If the default network net_attach_def does not exist, we'd need to create default OVN Controller based on config.
+	_, err := cm.InitDefaultController(nil)
+	if err != nil {
+		return err
+	}
+
+	hasSCTPSupport, err := util.DetectSCTPSupport()
+	if err != nil {
+		return err
+	}
+
+	if !hasSCTPSupport {
+		klog.Warningf("SCTP unsupported by this version of OVN. Kubernetes service creation with SCTP will not work ")
+	} else {
+		klog.Info("SCTP support detected in OVN")
+	}
+	cm.SCTPSupport = hasSCTPSupport
+
+	// enableOVNLogicalDataPathGroups sets an OVN flag to enable logical datapath
+	// groups on OVN 20.12 and later. The option is ignored if OVN doesn't
+	// understand it. Logical datapath groups reduce the size of the southbound
+	// database in large clusters. ovn-controllers should be upgraded to a version
+	// that supports them before the option is turned on by the master.
+	nbGlobal := nbdb.NBGlobal{
+		Options: map[string]string{"use_logical_dp_groups": "true"},
+	}
+	if err := libovsdbops.UpdateNBGlobalSetOptions(cm.nbClient, &nbGlobal); err != nil {
+		return fmt.Errorf("failed to set NB global option to enable logical datapath groups: %v", err)
+	}
+
+	metrics.RunTimestamp(cm.defaultStopChan, cm.sbClient, cm.nbClient)
+	metrics.MonitorIPSec(cm.nbClient)
+	if config.Metrics.EnableConfigDuration {
+		// with k=10,
+		//  for a cluster with 10 nodes, measurement of 1 in every 100 requests
+		//  for a cluster with 100 nodes, measurement of 1 in every 1000 requests
+		metrics.GetConfigDurationRecorder().Run(cm.nbClient, cm.kube, 10, time.Second*5, cm.defaultStopChan)
+	}
+
+	cm.podRecorder.Run(cm.sbClient, cm.defaultStopChan)
+
+	// Start and sync the watch factory to begin listening for events
+	if err := cm.watchFactory.Start(); err != nil {
+		return err
+	}
```

Comment:
This should be in a Run method int his controller instead of in a `Init` method that connotes initialization.